### ISSUE TITLE
🐛 Bug: Compliance bot isn't posting comments on PRs from forks

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/create-typescript-app#732]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

https://github.com/mtfoley/pr-compliance-action is used to verify that PRs match a set of rules. Those include having a title that confirms to conventional commits. If any failures are detected, it should post a GitHub comment on the PR.

### Actual

It's not :( https://github.com/JoshuaKGoldberg/create-typescript-app/pull/731 -> https://github.com/JoshuaKGoldberg/create-typescript-app/actions/runs/6038776422/job/16385877918?pr=731:

```plaintext
Warning: PR Body did not match required format
Error: This PR's title should conform to specification at https://conventionalcommits.org/
Error: Resource not accessible by integration
```

### Additional Info

I think I'd fixed this before. Need to check on that...

**Comment by JoshuaKGoldberg** [comment]
Maybe related: https://github.com/mtfoley/pr-compliance-action/issues/368
